### DISCUSSION TITLE
Keycloak getting error when try to login as "Invalid parameter: redirect_uri"

### DISCUSSION THREAD

**Discussion by upendraTekdi** [keycloak/keycloak#20233] [Q&A]
Keycloak getting error when try to login as "Invalid parameter: redirect_uri"
bydefault taking redirect_uri : http://localhost:3000/login
keycloak version : 20.0.0

`  let initOptions = {
      url: process.env.REACT_APP_REALM_KEYCLOAK_URL,
      realm: realmName,  // come from react state value
      clientId: process.env.REACT_APP_REALM_CLIENT_ID,
      onLoad: "login-required",
      KeycloakResponseType: "code",
    };
    alert(JSON.stringify(initOptions) + "working keycloak....");

    const keycloak = Keycloak(initOptions);
    alert(JSON.stringify(keycloak) + " keycloak....");

    keycloak
      .init({ onLoad: "check-sso", redirectUri: process.env.REACT_APP_REDIRECT_URI })
      .success((auth) => {
        if (!auth) {
          window.location.reload();
        } else {
          alert("Authenticated", keycloak.token);
        }`

**Comment by slominskir** [comment]
Google is your friend:  https://stackoverflow.com/questions/45352880/keycloak-invalid-parameter-redirect-uri

Tried any of those solutions yet?

**Reply by upendraTekdi** [reply]
yes i tried it .. but when i init and login on keyclock then i am getting error for access token  and keycloak data

error : post : "http://localhost:8082/auth/realms/Biogen/protocol/openid-connect/token" unautharized ( 401 error )  or An internal server error has occurred

**Reply by jonkoops** [reply]
Please refrain from mentioning project members randomly, we are already at capacity and spamming with more notifications is not gonna help.
